### PR TITLE
6123 hw info

### PR DESCRIPTION
**Pull request by konangc87**
# Hardware Info on sysinfo

## **Related Issue**
[6123](https://github.com/wazuh/wazuh/issues/6123)

## Description
This issue aims to refactor the hw info provider for each os using C++ reducing the scope and responsibilities of the modules..
These changes have to be made in the various implementations according to the operating system targeted for this feature.

## DoD / Tasks
- [X] Implementation
- [X] Run testtool on linux
image
- [X] Run testtool on windows
image
- [X] Run testtool on openbsd
image
- [X] Run testtool on freebsd
image
- [X] Run testtool on macos
image
- [X] RTR
image
- [X] SysInfo UT
image

### CODE REVIEW

File: src/shared_modules/utils/registryHelper.h [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+#include <winreg.h>
+#include <cstdio>
+#include <memory>
+	
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-function"
+
+namespace Utils
+{
+    class Registry
```
Comment (tinachoB) [inline src/shared_modules/utils/registryHelper.h:28]: final

File: src/shared_modules/utils/registryHelper.h [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+                {
+                    result,
+                    std::system_category(),
+                    "Error reading DWORD value of: " + valueName
+                };
+            }
+            return ret;
+        }
+
+        std::string string(const std::string& valueName)
```
Comment (tinachoB) [inline src/shared_modules/utils/registryHelper.h:58]: Could this method be const?

File: src/wazuh_modules/syscollector/sysInfoWin.cpp [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+                }
+            }
+        }
+    }
+    return ret;
+}
+
+std::string SysInfo::getCpuName()
+{
+    Utils::Registry reg(HKEY_LOCAL_MACHINE, "HARDWARE\\DESCRIPTION\\System\\CentralProcessor\\0");
```
Comment (tinachoB) [inline src/wazuh_modules/syscollector/sysInfoWin.cpp:137]: Maybe you can create a constexpr auto with this value and use it here (and in the line 142) and in the UTs.

File: src/wazuh_modules/syscollector/sysInfo.hpp [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+class SysInfo
+{
+public:
+	SysInfo() = default;
+	virtual ~SysInfo() = default;
+	nlohmann::json hardware();
+private:
+    virtual std::string getSerialNumber();
+    virtual std::string getCpuName();
```
Comment (tinachoB) [inline src/wazuh_modules/syscollector/sysInfo.hpp:23]: maybe these methods could be const

File: src/wazuh_modules/syscollector/sysInfoWin.cpp [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+    GetSystemInfo(&siSysInfo);
+    return siSysInfo.dwNumberOfProcessors;
+}
+void SysInfo::getMemory(nlohmann::json& info)
+{
+    MEMORYSTATUSEX statex;
+    statex.dwLength = sizeof(statex);
+    if (GlobalMemoryStatusEx(&statex))
+    {
+        info["ram_total"] = statex.ullTotalPhys/1024;
```
Comment (tinachoB) [inline src/wazuh_modules/syscollector/sysInfoWin.cpp:157]: Please, create a "constexpr auto KByte{1024};" as you did in sysInfoOpenBSD.cpp, line 18 and use it in both sides

File: src/wazuh_modules/syscollector/syscollectorImp.h [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+public:
+	Syscollector(const std::chrono::milliseconds& timeout);
+	void start();
+	~Syscollector();
+private:
+	void sync();
+	const std::chrono::milliseconds m_timeout;
+	std::atomic_bool				m_running;
+	std::thread 					m_thread;
+    SysInfo                         m_info;
```
Comment (tinachoB) [inline src/wazuh_modules/syscollector/syscollectorImp.h:30]: Please, fix indentation

File: src/wazuh_modules/syscollector/syscollectorImp.h [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+ */
+#ifndef _SYSCOLLECTOR_IMP_H
+#define _SYSCOLLECTOR_IMP_H
+#include <chrono>
+#include <thread>
+#include <atomic>
+#include "sysInfo.hpp"
+#include "json.hpp"
+
+class Syscollector
```
Comment (tinachoB) [inline src/wazuh_modules/syscollector/syscollectorImp.h:19]: final

**Review by tinachoB — CHANGES_REQUESTED** [review]

**Review by tinachoB — APPROVED** [review]
Looks great!

File: src/wazuh_modules/syscollector/sysInfoWin.cpp [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+        {
+            ret = "unknown";
+        }
+    }
+    else
+    {
+        const auto size {GetSystemFirmwareTable('RSMB', 0, nullptr, 0)};
+        if (size)
+        {
+            std::unique_ptr<unsigned char> buff{new unsigned char[size]};
```
Comment (Dwordcito) [inline src/wazuh_modules/syscollector/sysInfoWin.cpp:120]: std::make_unique(size)

**Review by Dwordcito — CHANGES_REQUESTED** [review]

**Review by tinachoB — APPROVED** [review]

**Review by Dwordcito — APPROVED** [review]

**Pull request merged**